sdk/core/azure-core/inc/azure/core/exception.hpp

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Define RequestFailedException. It is used by HTTP exceptions.
```

Comment:
Docs saym, "It is used by HTTP exceptions." Shouldn't it be in Http namespace then? Also, if it is used by Http Requests, DateTime should not be throwing it; and even for auth - not all cases when we throw are due to HTTP/network/requests, sometimes things are misconfigured (you create environment credential, but no environment variables are defined).

I think what you are trying to create, is Azure::Core::Exception, and then DateTime and Auth exceptions derive from it, and so should Azure::Core:Http::RequestException.